crates/ntx-builder/src/transaction.rs

```diff
@@ -116,16 +116,16 @@ impl NtxContext {
             .set_attribute("reference_block.number", chain_tip_header.block_num());
 
         async move {
-            async move {
+            Box::pin(async move {
                 let data_store = NtxDataStore::new(account, chain_tip_header, chain_mmr);
 
                 let notes = notes.into_iter().map(Note::from).collect::<Vec<_>>();
                 let (successful, failed) = self.filter_notes(&data_store, notes).await?;
                 let executed = Box::pin(self.execute(&data_store, successful)).await?;
-                let proven = Box::pin(self.prove(executed)).await?;
-                self.submit(proven).await?;
+                let proven = Box::pin(self.prove(executed.clone())).await?;
+                Box::pin(self.submit(proven, executed)).await?;
```

Comment:
@bobbinth LMK if you have any better ideas for this. Might be worth updating base to have `impl From<ProvenTransaction> for ExecutedTransaction` which can be called deeper in the stack with a feature flag. That way we don't affect the `submit()` API or this part of the stack.